app/scripts/modules/core/src/state/recoilStateInitializer.ts

```diff
@@ -0,0 +1,11 @@
+import { MutableSnapshot } from 'recoil';
+
+import * as atoms from './recoilAtoms';
+import { CollapsibleSectionStateCache } from '../cache/collapsibleSectionStateCache';
+
+export const initializeRecoilState = ({ set }: MutableSnapshot): void => {
```

Comment:
I don't think we need this `initializeRecoilState`.    What is the purpose?  This seems like it would become an unmaintainable mess of state objects.  AFAICT the atom already manages its default value internally?

Reply:
The docs mention that there is a period of time where the state is unset before initializing all the atoms if we do not use this: https://recoiljs.org/docs/guides/persistence/#restoring-state

`Note that it is important to use this prop instead of just manually setting atom values in an effect. Otherwise there will be an initial render without the restored state which can cause flicker or be invalid.`

I don't think everything needs this, but felt for this case it was important so that the navigation doesn't flicker.

Reply:
> Note that it is important to use this prop instead of just manually setting atom values in an effect. Otherwise there will be an initial render without the restored state which can cause flicker or be invalid.

But you're not doing what the docs warn against (setting the recoil state in a `useEffect`). Did you observe it actually flickering before hand?  The warning in the docs is against reading and rendering a _stale value_ in the first render, but the value in the first render should be the default value.

Reply:
Not yet. I haven't implemented any atom functionality, since I wanted this PR to be about the setup. I will just remove for now until I am actually utilizing this first atom.